components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice_test.go

```diff
@@ -0,0 +1,406 @@
+                        Passed:        false,
+                        SuccessMsg:    "",
+                        ErrorMsg:      "wrongcammand is not available",
+                        ResolutionMsg: "Ensure wrongcammand is available in $PATH on the node",
+                    },
+                    {
+                        Title:         "Ubuntu availability",
+                        Passed:        true,
+                        SuccessMsg:    "Ubuntu version is 20.04",
+                        ErrorMsg:      "",
+                        ResolutionMsg: "",
+                    },
+                },
+            },
+            expectedError: nil,
+        },
+        {
+            description: "If we are not able to get the OS Version",
+            args: args{
+                query:      "postgres",
+                checkarray: checktrue,
+                osFilepath: failfilepath,
+            },
+            expectedBody: models.SoftwareVersionDetails{
+                Passed: false,
+                Checks: []models.Checks{
+                    {},
+                },
+            },
+            expectedError: errors.New("open /failfilepath: no such file or director"),
+        },
+		{
+			description: "If the os version is not supported by automate",
+			args: args{
+				query: "postgres",
+				checkarray: checktrue,
+				osFilepath: failurefile,
+			},
+			expectedBody: models.SoftwareVersionDetails{
+				Passed: false,
+				Checks: []models.Checks{
+					{
+						Title:         "stat availability",
+                        Passed:        true,
+                        SuccessMsg:    "stat is available",
+                        ErrorMsg:      "",
+                        ResolutionMsg: "",
+					},
+					{
+						Title:         "mkdir availability",
+                        Passed:        true,
+                        SuccessMsg:    "mkdir is available",
+                        ErrorMsg:      "",
+                        ResolutionMsg: "",
+					},
+					{
+						Title:         "Debian GNU/Linux 10 (buster) availability",
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      "Debian GNU/Linux 10 (buster) version is not supported by automate",
+						ResolutionMsg: "Ensure Debian GNU/Linux 10 (buster) correct version is installed on the node",
+					
+					},
+				},
+			},
+		},
+    }
+    for _, tt := range tests {
+        t.Run(tt.description, func(t *testing.T) {
+            service.(*SoftwareVersionService).osFilepath = tt.args.osFilepath
+            service.(*SoftwareVersionService).cmdCheckArray = tt.args.checkarray
+            got, err := service.GetSoftwareVersionServices(tt.args.query)
+            if err != nil {
+                logger.NewLogrusStandardLogger().Error(err)
+            }
+            if !reflect.DeepEqual(got, tt.expectedBody) {
+                t.Errorf("Get Softwarever = %v want %v", got, tt.expectedBody)
+            }
```

Comment:
We can use assert.NoErr to check error, to compare we can use assert.Equal

Reply:
Done